pkg/common/uptime/uptime_test.go

```diff
@@ -0,0 +1,39 @@
+package uptime
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/spiffe/spire/test/fakes/fakemetrics"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestUptime(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+
+	for _, tt := range []struct {
+		name            string
+		reportInterval  time.Duration
+		expectedMetrics []fakemetrics.MetricItem
+	}{
+		{
+			name:           "report uptime metrics with 10 millisecond internal",
+			reportInterval: 10 * time.Millisecond,
+			expectedMetrics: []fakemetrics.MetricItem{
+				{Type: fakemetrics.SetGaugeType, Key: []string{"uptime_in_ms"}, Val: 10},
+			},
+		},
+	} {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			ctx, cancel := context.WithCancel(ctx)
+			defer cancel()
+			metrics := fakemetrics.New()
+			ReportMetrics(ctx, tt.reportInterval, metrics)
+			time.Sleep(tt.reportInterval)
```

Comment:
rather than sleep in a test, may be better to make use of use of a mock clock to be inserted into the reporting cadence

Reply:
updated with clock mock.